### ISSUE TITLE
MapsManager error

### ISSUE THREAD

**Issue by azuspl** [Flatracer/DemolitionDerbyGamemode#13]
Hey i have issue with gamemode

```
>> Demolition Derby Gamemode:

>>> Checking for the folder "DemolitionDerbyMaps"!

>>> The folder "DemolitionDerbyMaps" does exist!
 

>>> Getting Maps now!

 

Error resuming coroutine: SERVER/MapsManager.lua:13: attempt to index a nil value (local 'File')

stack traceback:

	SERVER/MapsManager.lua:13: in upvalue 'GetMapsFromPath'

	SERVER/MapsManager.lua:40: in upvalue 'CheckForDefaultMap'

	SERVER/MapsManager.lua:59: in upvalue 'CheckForMapsFolder'

	SERVER/MapsManager.lua:64: in function <SERVER/MapsManager.lua:63>

 
```

**Comment by azuspl** [comment]
Problem is with this code:

```
local function GetMapsFromPath()
	print('>>> Getting Maps now!\n')
	Content = ''
	local TMPFile = os.tmpname()
	if os.getenv('HOME') then
		local Result = os.execute('ls -a1 DemolitionDerbyMaps >' .. TMPFile)
	else
		local Result = os.execute('dir "DemolitionDerbyMaps" /b >' .. TMPFile)
	end
	local File = io.open(TMPFile, 'r')
	Content = File:read('*a')
	File:close()
	os.remove(TMPFile)
	local ContentSplitted = StringSplit(Content, '\n')
	for Index, Value in ipairs(ContentSplitted) do
		if Value and Value ~= '.' and Value ~= '..' and Value ~= '' then
			local XMLStart, XMLFinish = Value:lower():find('xml')
			if XMLStart and XMLFinish then
				table.insert(Maps, Value:sub(1, XMLFinish))
			end
		end
	end
end
```

**Comment by azuspl** [comment]
Okay i now where's the problem.
I do something like this:

	local File,err = io.open(TMPFile, 'r')
	if not File then return print(err) end
and now have "C:\Users\TC2E64~1\AppData\Local\Temp\s1az4.0: No such file or directory"

Some hostings blocking these Temp locale so it can't load these files where's map name's be.

**Comment by Flatracer** [comment]
Replace this:

```
if os.getenv('HOME') then
	local Result = os.execute('ls -a1 DemolitionDerbyMaps >' .. TMPFile)
else
	local Result = os.execute('dir "DemolitionDerbyMaps" /b >' .. TMPFile)
end
```

with this:

```
if os.getenv('HOME') then
	print('Linux')
	local Result = os.execute('ls -a1 DemolitionDerbyMaps >' .. TMPFile)
	print(Result)
else
	print('Windows')
	local Result = os.execute('dir "DemolitionDerbyMaps" /b >' .. TMPFile)
	print(Result)
end
```

And tell me what it outputs.

**Comment by azuspl** [comment]
```
>> Demolition Derby Gamemode:
>>> Checking for the folder "DemolitionDerbyMaps"!
>>> The folder "DemolitionDerbyMaps" does exist!
>>> Getting Maps now!
Windows
nil

Current version: 5.0.3
Updater version: 5.0.3
Everything is fine!
```

And it's all in the console.
But now i fixed it, just changed 
local TMPFile = 'DemolitionDerbyMaps/stmp.txt'
created in folder DemolitionDerbyMaps where's xml maps txt file named stmp and in this file i put maps names what i actually have:
Default Map 1.xml
Default Map 2.xml
Default Map 3.xml
Default Map 4.xml
Default Map 5.xml
[dd]cross28.xml
[dd]hyperlife2.xml

and removed:
os.remove(TMPFile)

And now server reading from this file
If you want i can send you my created two maps ^^


Actually code looks like this:
```
local function GetMapsFromPath()
	print('>>> Getting Maps now!\n')
	Content = ''
	local TMPFile = 'DemolitionDerbyMaps/stmp.txt'
if os.getenv('HOME') then
	print('Linux')
	local Result = os.execute('ls -a1 DemolitionDerbyMaps >' .. TMPFile)
	print(Result)
else
	print('Windows')
	local Result = os.execute('dir "DemolitionDerbyMaps" /b >' .. TMPFile)
	print(Result)
end
	local File,err = io.open(TMPFile, 'r')
	if not File then return print(err) end
	Content = File:read('*a')
	File:close()
	local ContentSplitted = StringSplit(Content, '\n')
	for Index, Value in ipairs(ContentSplitted) do
		if Value and Value ~= '.' and Value ~= '..' and Value ~= '' then
			local XMLStart, XMLFinish = Value:lower():find('xml')
			if XMLStart and XMLFinish then
				table.insert(Maps, Value:sub(1, XMLFinish))
			end
		end
	end
end
```

Thanks for help and for creating this gamemode in FiveM.. FiveM needs more than RP, good work with this gamemode! ^^
Still remember MTA gamemodes like DD/DM.

**Comment by Flatracer** [comment]
>  Still remember MTA gamemodes like DD/DM.

Thats why I accepted to create this with Scotty when he asked me ^^

About the Map. Would be nice if you could upload them here.

But be aware of the need of 32 vehicles in the map. Otherwise it won't work.

**Comment by azuspl** [comment]
Yeah i added 32 spawns for cars. But i have a question what map editor you use? Beacuse my xml maps is other like yours and i have small problem with repair pickups, added as a pickup in map editor, saved as a xml, added to server and my maps work, yes but these repair pickups dissapear, but if i load this map in map editor again these repair pickups still be here

There's maps:
https://transfer.sh/YWX6f/mapgi.zip

**Comment by Flatracer** [comment]
I use Menyoo.

Pickups are currently only officially supported when using Menyoo Maps but by checking your map I will add the support for Map Editor Maps too.

**Comment by Flatracer** [comment]
@azuspl Which pickup did you add in which map?

**Comment by azuspl** [comment]
Repair pickup on these two maps

**Comment by Flatracer** [comment]
Which of these:

PROP_MK_REPAIR
PROP_MP_REPAIR
PROP_MP_REPAIR_01

**Comment by azuspl** [comment]
PROP_MP_REPAIR

**Comment by azuspl** [comment]
Ooo, good job and nice car :D

**Comment by azuspl** [comment]
![unknown](https://user-images.githubusercontent.com/30390034/44307450-26c15000-a3a3-11e8-928e-04d8214562d7.png)

Actually created DD server, now i talking with youtubers, maybe they will play on Live's
Hope this gamemode will be popular in fivem ^^

**Comment by ScottUK** [comment]
Feel free to push your maps to the resource as a pull request

**Comment by azuspl** [comment]
Okay, thanks i will add there new maps if i finish them ^^

And have small question. Where leaderboard saving/load scores? Beacuse on my server it not work ;x (I downloaded your update.) 
Maybe can you add saving as a txt file in demolitionderby folder for leaderboards? Maybe it's a better option beacuse some server hostings don't give permissions to save files in temp folders

**Comment by Flatracer** [comment]
Leaderboards are saved in the resources rolder. There should be a folder called "Leaderboards"

**Comment by Flatracer** [comment]
An update in the future will add more balanced vehicle selection.

**Comment by azuspl** [comment]
1. Boost working for all players (Maybe it's my server problem?)
2. If players disconnect/timed out his car no dissapear
3. Spectators is under my car 
4. As a spectator controls not work like a ESC etc..
![doneee](https://user-images.githubusercontent.com/30390034/44312177-3a48d700-a3f4-11e8-8ab0-ee18f200e5bd.png)

**Comment by Flatracer** [comment]
Fixed with commit 4d46bc2
